ray_lightning/ray_ddp.py

```diff
@@ -530,21 +182,34 @@ def world_size(self) -> int:
     def local_rank(self) -> int:
         return self._local_rank
 
+    @local_rank.setter
+    def local_rank(self, value: int):
+        self._local_rank = value
+
     @property
     def global_rank(self) -> int:
         return self._global_rank
 
+    @global_rank.setter
+    def global_rank(self, value: int):
+        self._global_rank = value
+
     @property
     def node_rank(self) -> int:
         return self._node_rank
 
     @property
     def root_device(self):
+        if self._device:
+            return self._device
         if self.use_gpu and torch.cuda.is_available():
             if self._is_remote:
                 # Adjust to support multiple GPUs per worker or fractional
                 # GPUs per worker.
-                device_id = ray.get_gpu_ids()[0]
+                gpu_id = ray.get_gpu_ids()[0]
+                cuda_visible_list = list(
+                    map(int, ray._private.utils.get_cuda_visible_devices()))
```

Comment:
instead of using a private api can we read directly from CUDA_VISIBLE_DEVICES env var here?

Reply:
oh, my bad